File: Subject Archive/INFO1910 Intro to programming/weekly_tasks/wk11/fib/fib.py
import sys
def fib(x):
    if x == 0:
        return 0

    total = 1
    prev = 0

    while x > 0:
        temp = total
        total += prev
        prev = temp
        x -= 1

    return total

print(fib(int(sys.argv[1])))

File: Subject Archive/INFO1910 Intro to programming/transpiler/test/linker/case_2/fib.c
#include "lib/fib.h"

float fibonacci (int length){
    float prev = 0;
    float cur = 1;
    length -= 2;

    printf("%.0f\t", prev);
    printf("%.0f\t", cur);
    while (length > 0)
    {
        int temp = cur;
        cur += prev;
        prev = temp;
        length = length -1;
        printf("%.0f\t", cur);
    }

    printf("\n");

    float gr = cur/prev;

    return gr;
}